src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -8,6 +8,46 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
 {
     public class UninitializedNonNullableFieldTests : CSharpTestBase
     {
+        [Fact]
+        public void UninitializedEvents()
+        {
+            var src = @"
+using System;
+class C
+{
+    public event Action<object?> E1;
+    public event Action<object?> E2 { add { } remove { } }
+#pragma warning disable 0414
+    public event Action<object?> E3 = null!;
+    public event Action<object?>? E4 = null;
+#pragma warning restore 0414
+
+    internal C()
+    {
+    }
+
+    internal C(Action<object?> e)
+    {
+        E1 = e;
+        E2 += e;
+    }
+
+    internal C(object o)
+    {
+        E1 += p => {};
+        E2 += p => {};
+    }
+}";
+            var comp = CreateCompilation(src, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                    // (12,14): warning CS8618: Non-nullable event 'E1' is uninitialized.
+                    //     internal C()
+                    Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("event", "E1").WithLocation(12, 14),
+                    // (22,14): warning CS8618: Non-nullable event 'E1' is uninitialized.
+                    //     internal C(object o)
```

Comment:
>internal C(object o) [](start = 27, length = 20)

`E1` is initialized with `+=`.

Reply:
Right now we never respect `+=` for accessibility. This PR confirms that behavior.

Reply:
Just changed this to implement the += behavior.